pkg/controller/federatedtypeconfig/controller.go

```diff
@@ -50,8 +50,9 @@ type Controller struct {
 	client genericclient.Client
 
 	// Map of running sync controllers keyed by qualified target type
-	stopChannels map[string]chan struct{}
-	lock         sync.RWMutex
+	stopChannels      map[string]chan struct{}
+	syncerGenerations map[string]int64
```

Comment:
As per my previous review, having to track the generation here instead of in the API seems unnecessary to me. If `ObservedGeneration` and `Generation` should not always be the same, and that should allow the avoidance of maintaining generation state on the controller.

Reply:
@marun this is only for test purpose. It is not used for functions.

Currently, I only check `ObservedGeneration` and `Generation` as you proposed. In order to check if the controller refreshed in test case, I add this one to trace if a controller for a specific generation is updated. Or any idea from your side to test a refreshed controller?

Reply:
I think test-only in-memory state should only exist if the controller is under test. As proposed, the state would be maintained even if the controller were running in a non-test context.

That said, tracking the generation in-memory does not appear to be necessary. The same information is provided via `ObservedGeneration` on the FTC. It should be sufficient to ensure that `ObservedGeneration` is only updated if the sync controller for the FTC reflects the latest generation.